cli/find.php

```diff
@@ -111,27 +108,53 @@
     fputcsv($fp, ['Table', 'Column', 'courseid', 'idnumber']);
 }
 
-// Output the result.
-foreach ($result as $table => $columns) {
-    foreach ($columns as $column => $rows) {
+// Perform the search.
+$searchlist = helper::build_searching_list($tables);
+
+// Output the result for each table.
+foreach ($searchlist as $table => $columns) {
+    // Summary mode.
+    $limit = $options['summary'] ? 1 : 0;
+
+    foreach ($columns as $column) {
+        // Show the table and column being searched.
+        $colname = $column->name;
+        echo "Searching in table: $table, column: $colname\n";
+
+        // Perform the search.
+        if (!empty($options['regex-match'])) {
+            $result = helper::regular_expression_search($search, $table, $column, $limit);
+        } else {
+            $result = helper::plain_text_search($search, $table, $column, $limit);
```

Comment:
We never want to return results and pass them around. Instead have $fp as an argument and have plain_text_search and regular_expression_search write directly to the file handle line by line as its finds the records